specification/at_protocol_specification.md

```diff
@@ -0,0 +1,1110 @@
+   <td>public:signing_publickey@<atsign>
+   </td>
+   <td>Public key used on a pol handler to verify a signed challenge
+   </td>
+  </tr>
+  <tr>
+   <td><atsign>@signing_privatekey@<atsign>
+   </td>
+   <td>Private key used to sign a challenge on a pol request
+   </td>
+  </tr>
+  <tr>
+   <td><atsign>:shared_key@<atsign>
+   </td>
+   <td>Symmetric key used to encrypt/decrypt self atsign data
+   </td>
+  </tr>
+</table>
+
+## Configuration Parameters
+A Secondary Server should honor the following configuration parameters.
+
+<!-- <TO DO: Ideally these parameters should be set using some verb so that the user can control them. config verb can be extended to accomplish the same. Config verb also has the list option, so we can set/reset/list the configuration using the same> -->
+
+<table>
+  <tr>
+   <td>
+   <strong>Key</strong>
+   </td>
+   <td><strong>Valid Values</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><strong>autoNotify</strong>
+   </td>
+   <td>true/false
+   </td>
+   <td>If set to true, a Secondary Server should automatically notify another @sign user when a key has been shared with him/her. Please refer to the <em><span style="text-decoration:underline;">notify </span></em>verb spec for details.
+   </td>
+  </tr>
+  <tr>
+   <td><strong>bufferLimit </strong>
+   </td>
+   <td>Number of bytes
+   </td>
+   <td>Maximum size of a value for a key that can be transferred to a Secondary Server
+   </td>
+  </tr>
+  <tr>
+   <td><strong>inbound_max_limit</strong>
+   </td>
+   <td>An Integer
+   </td>
+   <td>Maximum number of inbound connections that a Secondary Server can accept
+   </td>
+  </tr>
+  <tr>
+   <td><strong>outbound_max_limit</strong>
+   </td>
+   <td>An Integer
+   </td>
+   <td>Maximum number of outbound connections that a secondary can make to another Secondary Server
+   </td>
+  </tr>
+  <tr>
+   <td><strong>inbound_idle_time_millis</strong>
+   </td>
+   <td>Time in milliseconds
+   </td>
+   <td>Maximum time the inbound connection can be active
+   </td>
+  </tr>
+  <tr>
+   <td><strong>outbound_idle_time_millis</strong>
+   </td>
+   <td>Time in milliseconds
+   </td>
+   <td>Maximum time the outbound connection can be active.
+   </td>
+  </tr>
+</table>
+
+## Block List
+A user of the Secondary Server should be able to decide who is allowed to connect to a Secondary Server. The `config` verb should be used to configure this. Once added, a Secondary Server should honor the list at the time of accepting new connections from an @sign user using the `from` verb.
+
+<!-- ## Life Cycle -->
+
+<!-- <TO DO: TBD: What I want to write is, A User of the secondary should be able to control pausing and resuming of a Secondary Server> -->
+
+## Verbs
+
+### The `from` verb
+
+**Synopsis:**
+
+The `from` verb is used to tell a secondary whom you claim to be.
+
+Following regex represents the syntax of the `from` verb:
+
+**Example:**
+
+```r'^from:(?<@sign>@?[^@\s]+$)' ```
```

Comment:
Some people will **get** the regex, but many won't, so for each regex it would be useful to add an actual example or a set of typical examples.